salt/_macros/certs.jinja

```diff
@@ -120,3 +124,38 @@
         - x509: {{ key }}
 
 {%- endmacro %}
+
+# externally-sourced certs; content in a pillar
+{% macro external_pillar_certs(crt, crt_pillar, key, key_pillar, bundle=False) -%}
+{{ crt }}:
+  file.managed:
+    - user:  root
+    - group: root
+    - mode: 0644
+    - contents_pillar: {{ crt_pillar }}
+
+{{ key }}:
+  file.managed:
+    - user:  root
+    - group: root
+    - mode: 0444
+    - contents_pillar: {{ key_pillar }}
+
+{% if bundle %}
+# TODO: move this out to a macro so it's not repeated in here?
```

Comment:
While I agree with the nicety of 'hey someday we might want to make this cleaner' I doubt it will ever happen, so you might as well remove this TODO. Or leave it in -shrug-

Reply:
Leaving it in there in case we end up in a weird dystopian future where this codebase is the best path forward. ;)